assets/app/actions/siteActions.js

```diff
@@ -0,0 +1,71 @@
+import federalist from '../util/federalistApi';
+import github from '../util/githubApi';
+import { siteActionTypes } from '../constants';
+import store from '../store';
+
+export default {
+  fetchSites() {
+    federalist.fetchSites().then((sites) => {
+      store.dispatch({
+        type: siteActionTypes.SITES_RECEIVED,
+        sites
+      });
+    });
+  },
+
+  addSite(siteToAdd) {
+    federalist.addSite(siteToAdd).then((site) => {
+      store.dispatch({
+        type: siteActionTypes.SITE_ADDED,
+        site
+      });
+    });
+  },
+
+  deleteSite(siteId) {
+    federalist.deleteSite(siteId);
+  },
+
+  deletedSite(siteId) {
+    store.dispatch({
+      type: siteActionTypes.SITE_DELETED,
+      siteId
+    });
+  },
+
+  fetchSiteAssets(site) {
+    const config = site['_config.yml'];
+    const assetPath = (config && config.assetPath) || 'assets';
+    github.fetchRepositoryContent(site, assetPath).then((assets) => {
+      return assets.filter((asset) => {
+        return asset.type === 'file';
+      });
+    }).then((assets) => {
+      store.dispatch({
+        type: siteActionTypes.SITE_ASSETS_RECEIVED,
+        siteId: site.id,
+        assets
+      });
+
+      return Promise.resolve(site);
+    });
+  },
+
+  fetchSiteConfigs(site) {
+    return github.fetchRepositoryConfigs(site).then((configs) => {
+      store.dispatch({
+        type: siteActionTypes.SITE_CONFIGS_RECEIVED,
+        siteId: site.id,
+        configs
+      });
+
+      return Promise.resolve(site);
+    });
+  },
+
+  fetchSiteConfigsAndAssets(site) {
+    return this.fetchSiteConfigs(site).then((site) => {
```

Comment:
Well, right now the sites reducer acts ([reducers.js#L55-L59](https://github.com/18F/federalist/blob/react-redux/assets/app/reducers.js#L55-L59)) on `SITE_CONFIGS_RECEIVED` to update the respective site with the configuration files. We could have an event that is for both configs and assets, but I think its fine to keep them discreet states for now.